extensions/src/main/scala/water/parser/CategoricalPreviewParseWriter.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package water.parser;
+
+import java.lang.reflect.Field;
+import water.util.IcedHashMap;
+
+public class CategoricalPreviewParseWriter extends PreviewParseWriter {
+
+  public CategoricalPreviewParseWriter(String[] domain, int totalCount) {
+    super(1);
+    this._nstrings[0] = totalCount;
+    IcedHashMap<String, String>[] domains = new IcedHashMap[1];
+    domains[0] = new IcedHashMapWrapper(domain);
```

Comment:
The proxy/wrapper is fine for now. There is a clear reason for that - memory savings. The very fact that the original `PreviewParseWriter` uses HashMap as a set effectively doubles the already doubled memory (extreme case). 

My suggestion:

- Keep this as-is, as the internal logic of the `guessTypes` method seems to be compliant with this proxy implementation of IcedHashMap.
- In the very next H2O fix release, make changes to `PreviewParseWriter` to enabled better embedding in SW. This could also include migration to a much later introducerd IcedHashSet.
- Overridable method `getDomain()` should be introduced in the parent class as wel..